### ISSUE TITLE
Does splitting the input_ids into chunks make the results non-comparable?

### ISSUE THREAD

**Issue by ytdu** [Jellyfish042/uncheatable_eval#23]
>             neg_log_prob_temp = 0
>             for begin in range(0, seq_length, chunk_size):
>                 input_chunk = inputs["input_ids"][:, begin : begin + chunk_size]
>                 if bos_mode in ["add_default_bos", "add_default_eos", "add_newline_token"]:
>                     input_chunk = torch.cat([bos_tensor, input_chunk], dim=-1)
>                 if bos_mode in ["replace_with_bos", "replace_with_eos", "replace_with_newline_token"]:
>                     input_chunk[0, 0] = bos_token
>                 logit = model.forward(input_ids=input_chunk).logits[0, :, :]
>                 loss = self.calculate_log_sum(logit, input_chunk.squeeze(0))
>                 loss_sum = loss.sum().item()
>                 neg_log_prob_temp += loss_sum

Hi, I noticed that when calculating neg_log_prob, the evaluator splits input_ids into chunks.

Different models may have different tokenizers, so the chunking can vary from models to models.

I wonder if different chunking would lead to different scores so that the results are not comparable for different models.
